### PR TITLE
Replace SpanStatus.CanonicalCode with OTel status code

### LINKED ISSUE #29: Remove SpanStatus CanonicalCode
OpenTelemetry deprecated GRPC-like status codes a while ago. We might want to remove them from our API as well:

**Current spec**
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-status

**Change in `.proto` file**
https://github.com/open-telemetry/opentelemetry-proto/commit/59c488bfb8fb6d0458ad6425758b70259ff4a2bd#diff-5c407ac9c675a60170f94ab620f5329f25e5388d6790864b7799abce66f79332

*This commit also added backwards-compatibility notes we might want to follow.*

### PR DESCRIPTION
**Pull request by slashmo**
Closes #29

### CODE REVIEW

File: Sources/Tracing/Span.swift [hunk 1]
```diff
@@ -583,61 +583,29 @@ extension SpanAttributes: ExpressibleByDictionaryLiteral {
 // ==== ----------------------------------------------------------------------------------------------------------------
 // MARK: Span Status
 
-/// Represents the status of a finished Span. It's composed of a canonical code in conjunction with an optional descriptive message.
+/// Represents the status of a finished Span. It's composed of a status code in conjunction with an optional descriptive message.
 public struct SpanStatus {
-    public let canonicalCode: CanonicalCode
+    public let code: Code
```
Comment (ktoso) [inline Sources/Tracing/Span.swift:588]: Should it be possible to set one with undefined status code?

File: Sources/Tracing/Span.swift [hunk 2]
```diff
@@ -583,61 +583,29 @@ extension SpanAttributes: ExpressibleByDictionaryLiteral {
...
-        /// Operation is not implemented or not supported/enabled in this service.
-        case unimplemented
-        /// Internal errors. Means some invariants expected by underlying system has been broken.
-        case `internal`
-        /// The service is currently unavailable. This is a most likely a transient condition and may be corrected by retrying with a backoff.
-        case unavailable
-        /// Unrecoverable data loss or corruption.
-        case dataLoss
-        /// The request does not have valid authentication credentials for the operation.
-        case unauthenticated
```
Comment (ktoso) [inline Sources/Tracing/Span.swift:640]: Good riddance, was quite complex 👍

File: Sources/Tracing/Span.swift (hunk 1 above)
Comment (slashmo) [reply Sources/Tracing/Span.swift:588]: > Should it be possible to set one with undefined status code?
> @ktoso 

Good question ... I don't think so, as unset basically means you haven't altered the status. The only use-case would be to set it at first, then reset it afterwards. But I don't think that's a real concern.

File: Sources/Tracing/Span.swift (hunk 2 above)
Comment (slashmo) [reply Sources/Tracing/Span.swift:640]: Agreed 😊

**Review by ktoso — APPROVED** [review]

**Pull request merged**